tests/testthat/test-infection-integration.R

```diff
@@ -531,3 +560,195 @@ test_that('update_severe_disease renders with no infections', {
+                                            late_clinical_failure_prob = 0,
+                                            late_parasitological_prob = 0,
+                                            reinfection_prob = 0)
+  clinical_infections <- individual::Bitset$new(20)
+  timestep <- 5
+  events <- create_events(parameters)
+  variables <- list(
+    state = list(queue_update = mockery::mock()),
+    infectivity = list(queue_update = mockery::mock()),
+    drug = list(queue_update = mockery::mock()),
+    drug_time = list(queue_update = mockery::mock())
+  )
+  renderer <- individual::Render$new(timesteps = 10)
+  
+  treated <- calculate_treated(variables = variables,
+                               clinical_infections = clinical_infections,
+                               parameters = parameters,
+                               timestep = timestep,
+                               renderer = renderer)
+  
+  expect_identical(object = treated$size(), expected = 0, info = "Error: calculate_treated() returning non-zero number of treated individuals
+                 in the absence of clinically infected individuals")
+  expect_identical(object = treated$to_vector(), expected = numeric(0), info = "Error: calculate_treated() returning non-zero number of treated individuals
+                 in the absence of clinically infected individuals")
+})
+
+test_that('calculate_treated() returns an empty Bitset when the parameter list contains no clinical
+          treatment or resistance parameters', {
+            parameters <- get_parameters()
+            clinical_infections <- individual::Bitset$new(20)$insert(1:20)
+            timestep <- 5
+            events <- create_events(parameters)
+            variables <- list(
+              state = list(queue_update = mockery::mock()),
+              infectivity = list(queue_update = mockery::mock()),
+              drug = list(queue_update = mockery::mock()),
+              drug_time = list(queue_update = mockery::mock())
+            )
+            renderer <- individual::Render$new(timesteps = 10)
+            
+            treated <- calculate_treated(variables = variables,
+                                         clinical_infections = clinical_infections,
+                                         parameters = parameters,
+                                         timestep = timestep,
+                                         renderer = renderer)
+            
+            expect_identical(object = treated$size(), expected = 0, info = "Error: calculate_treated() returning non-zero number of treated individuals
+                 in the absence of clinical treatment or resistance parameters")
+            expect_identical(object = treated$to_vector(), expected = numeric(0), info = "Error: calculate_treated() returning non-zero number of treated individuals
+                 in the absence of clinical treatment or resistance parameters")
```

Comment:
```suggestion
```

Reply:
I've removed the unnecessary additional test and left the following test:

expect_identical(object = treated$size(), expected = 0, info = "Error: calculate_treated() returning non-zero number of treated individuals in the absence of clinical treatment or resistance parameters")